sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -777,7 +791,7 @@ def arrow_schema_to_bq_schema(arrow_schema: pyarrow.Schema) -> List[SchemaField]
  Compute a deterministic hash for the `left_table_query_string` that will be used throughout
  all the logic as the field to GROUP BY the data
 */
-WITH entity_dataframe AS (
+CREATE TEMP TABLE entity_dataframe AS (
```

Comment:
is the main change here to create temp tables for everything? (and everything else got indented)?

Reply:
yes, create temp table for entity dataframe and every joining feature views.

Reply:
and the temp tables only exist within the current execution session

Reply:
Are you able to confirm what the scope of the session is? Does each invocation of `self._execute_query` create a new session? At a cursory glance, I see that we're not explicitly passing `create_session=True` in the `QueryJobConfig` instance. I think it would be very helpful to have a confirmation that there isn't going to be any cross contamination between concurrent historical fetches.

Reply:
`Does each invocation of self._execute_query create a new session?` yes it does

`I see that we're not explicitly passing create_session=True in the QueryJobConfig instance.` we only need to declare this config when we need to reuse the session in a different code scope. In this case, we gather all of feature views in a collection of scripts to execute once. I can confirm after the scripts are executed, temp tables do not exist anywhere else.